xknx/devices/climate_mode.py

```diff
@@ -125,30 +129,43 @@ def __init__(
         )
 
         self.operation_mode = HVACOperationMode.STANDBY
+        self.controller_mode = HVACOperationMode.HEAT
 
-        self.operation_modes_ = []
+        self._operation_modes = []
         if operation_modes is None:
-            self.operation_modes_ = self.gather_operation_modes()
+            self._operation_modes = self.gather_operation_modes()
         else:
             for mode in operation_modes:
                 if isinstance(mode, str):
-                    self.operation_modes_.append(HVACOperationMode(mode))
+                    self._operation_modes.append(HVACOperationMode(mode))
                 elif isinstance(mode, HVACOperationMode):
-                    self.operation_modes_.append(mode)
-
-        self.supports_operation_mode = (
-            group_address_operation_mode is not None
-            or group_address_operation_mode_state is not None
-            or group_address_operation_mode_protection is not None
-            or group_address_operation_mode_night is not None
-            or group_address_operation_mode_comfort is not None
-            or group_address_operation_mode_standby is not None
-            or group_address_controller_status is not None
-            or group_address_controller_status_state is not None
-            or group_address_controller_mode is not None
-            or group_address_controller_mode_state is not None
-            or group_address_heat_cool is not None
-            or group_address_heat_cool_state is not None
+                    self._operation_modes.append(mode)
+
+        self._controller_modes = []
+        if controller_modes is None:
+            self._controller_modes = self.gather_controller_modes()
+        else:
+            for mode in controller_modes:
+                if isinstance(mode, str):
+                    self._controller_modes.append(HVACOperationMode(mode))
```

Comment:
I agree that it's not nice and we should change it, but since all DPTs use `HVACOperationMode` I believe we should not change this in the scope of this PR. I'll create follow up issue for this.